tests/publiccloud/ssh_interactive_end.pm

```diff
@@ -16,7 +16,6 @@ use utils;
 sub run {
     my ($self, $args) = @_;
     select_host_console(force => 1);
-    $args->{my_provider}->cleanup($args) unless check_var('PUBLIC_CLOUD_SLES4SAP', 1);
```

Comment:
This one has been introduced for mr_test

Reply:
Who is Mr Test?

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/schedule/sles4sap/sles4sap_gnome_saptune.yaml

Reply:
Aha, it's only for SAP. I'll put it back then.

Reply:
wait , but it is saying **unless** check_var('PUBLIC_CLOUD_SLES4SAP', 1) . so if it is something SAP related check_var will return TRUE and unless will skip this line ! How it can be "for mr_test" ? :thinking:

Reply:
I agree with Anton. Michele can you please re-check this code and then let me know if I can remove this line?

Reply:
We actually DON'T want to call cleanup in case of mr_test, because we are running our own cleanup module after `ssh_interactive_end` (which also deals with peering), so AFAICT this is the correct behavior. 

Question is, removing this line would remove the cleanup from your tests as well (at least the cleanup at the end of ssh_interactive_end). If you actually intend to remove cleanup from your tests, then removing this line is fine (this line already skipped it for us). Right, @mpagot ?

Reply:
we NOT planning to remove the cleanup from our tests but AFAIK this call is just leftover which is left after we **moved** responsibility for cleanup out from `ssh_interactive_end` ( currently we using different mechanism to make sure that `terraform destroy` always called )

Reply:
TD;LR **NO ONE need this line hence it can and must be dropped** . ( @mpagot  please raise your hand if we all still missing something obvious :thinking:  )

Reply:
Yes, that's what I meant to say - if you plan to remove cleanup from this module, then removing it completely is the way to go. We also have our own dedicated cleanup module running after this

Reply:
Doing a final check and confirming (unless Michele answers before me)

Reply:
Line was like "Do not perform cleanup if SLES4SAP". Now it become "Never perform cleanup here".
As far as I understand it is possible as responsibility has been moved in another place. 
My initial comment was not clear about it: what I mean is just "I remember this exception was specifically added for mr_test from all sles4sap: please run VR for mr_test to ensure that moved responsibility behave the same as before in mr_Test scheduling". I'm fine with the change, please run VR.

Reply:
@mpagot I'll be happy to run VR. Please provide link to clone.
I so far did:
* [sles4sap_gnome_saptune_delete_rename@ec2_r5b.metal](https://pdostal-server.suse.cz/tests/7839)	[15](https://pdostal-server.suse.cz/tests/7839)
* [sles4sap_gnome_saptune_delete_rename@az_Standard_E4s_v3](https://pdostal-server.suse.cz/tests/7840)	[18](https://pdostal-server.suse.cz/tests/7840)
* [SAPHanaSR-ScaleUp-PerfOpt@az_Standard_E4s_v3](https://pdostal-server.suse.cz/tests/7838)